src/creds.ts

```diff
@@ -0,0 +1,248 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import { handleError, konsola } from './utils'
+import chalk from 'chalk'
+
+export interface NetrcMachine {
+  login: string
+  password: string
+  region: string
+}
+
+export const getNetrcFilePath = () => {
+  const homeDirectory = process.env[
+    process.platform.startsWith('win') ? 'USERPROFILE' : 'HOME'
+  ] || process.cwd()
+
+  return path.join(homeDirectory, '.netrc')
+}
+
+const readNetrcFileAsync = async (filePath: string) => {
+  return await fs.readFile(filePath, 'utf8')
+}
+
+const preprocessNetrcContent = (content: string) => {
+  return content
+    .split('\n')
+    .map(line => line.split('#')[0].trim())
+    .filter(line => line.length > 0)
+    .join(' ')
+}
+
+const tokenizeNetrcContent = (content: string) => {
+  return content
+    .split(/\s+/)
+    .filter(token => token.length > 0)
+}
+
+const parseNetrcTokens = (tokens: string[]) => {
+  const machines: Record<string, NetrcMachine> = {}
+  let i = 0
+
+  while (i < tokens.length) {
+    const token = tokens[i]
+
+    if (token === 'machine' || token === 'default') {
+      const machineName = token === 'default' ? 'default' : tokens[++i]
+      const machineData: Partial<NetrcMachine> = {}
+      i++
+
+      while (
+        i < tokens.length
+        && tokens[i] !== 'machine'
+        && tokens[i] !== 'default'
+      ) {
+        const key = tokens[i] as keyof NetrcMachine
```

Comment:
@Edo-San I see few points on your feedback regarding type casting? Any particular reason you prefer not to cast types?

Reply:
@alvarosabu Type casting is very convenient, but it's also inherently wrong most of the time.
When you type cast a variable, you are telling the TS compiler that you are absolutely sure that a certain variable will have a specific object shape. But realistically you cannot be sure about it.

Random example related to this use case: what if we wrote a key the wrong way somewhere? Maybe we wrote `defautl` instead of `default`. Or maybe all the keys are right now, but in the future there could be typos anywhere. If this happens and you type cast, the compiler won't complain, and you're going to get a runtime error (or even worse it could fail without even throwing, if we're accessing an undefined object property).

The right way to check things is.. to check them! 😄
And Typescript does just that: it tells you when you need to check your data.